input/postgres/collection_helper.go

```diff
@@ -67,5 +92,69 @@ func NewCollectionHelper(ctx context.Context, logger *util.Logger, server *state
 		Roles:                     roles,
 		ConnectedAsSuperUser:      connectedAsSuperUser,
 		ConnectedAsMonitoringRole: connectedAsMonitoringRole,
+		HelperFunctions:           helpersFromFunctions(helperFunctions),
 	}, nil
 }
+
+func (s CollectionHelper) ForCurrentDatabase(functions []state.PostgresFunction) CollectionHelper {
+	return CollectionHelper{
+		Config:                    s.Config,
+		Logger:                    s.Logger,
+		SelfTest:                  s.SelfTest,
+		GlobalOpts:                s.GlobalOpts,
+		PostgresVersion:           s.PostgresVersion,
+		Roles:                     s.Roles,
+		ConnectedAsSuperUser:      s.ConnectedAsSuperUser,
+		ConnectedAsMonitoringRole: s.ConnectedAsMonitoringRole,
+		HelperFunctions:           helpersFromFunctions(functions),
+	}
+}
+
+func (s CollectionHelper) findHelperFunction(name string, inputTypes []string) (state.PostgresFunction, bool) {
+	funcs, exists := s.HelperFunctions[name]
+	if !exists {
+		return state.PostgresFunction{}, false
+	}
+	for _, f := range funcs {
+		args := strings.Split(f.Arguments, ", ")
+		if len(inputTypes) > len(args) {
+			// We're expecting more arguments than the function has
+			continue
+		}
+		mismatch := false
+		for idx, arg := range args {
+			// Split by this assumed pattern: "NAME TYPE DEFAULT DEFAULT_VALUE"
+			//
+			// Note this currently does not handle data types with spaces in
+			// them, such as "double precision", or other cases not expected
+			// with our known set of helper functions.
+			parts := strings.Split(arg, " ")
+			if idx >= len(inputTypes) && parts[2] != "DEFAULT" {
+				// Function has more arguments required than we're expecting
+				mismatch = true
+				break
+			}
+			if parts[1] != inputTypes[idx] {
+				mismatch = true
+				break
+			}
+		}
+		if !mismatch {
+			return f, true
+		}
+	}
+	return state.PostgresFunction{}, false
+}
+
+func (s CollectionHelper) HelperExists(name string, inputTypes []string) bool {
```

Comment:
For context, I plan to add additional behavior here soon that warns the user (and flags the SelfTest logic) in case the function ownership is not as expected. To keep reviews easy I've not yet added this here.